packages/playground/blueprints/src/lib/utils/logger.ts

```diff
@@ -0,0 +1,140 @@
+import { PHPRequestEndEvent } from "@php-wasm/universal/src/lib/universal-php";
+import { UniversalPHP } from "@php-wasm/universal/src/lib/universal-php";
+/**
+ * Log severity levels.
+ */
+export enum LogSeverity {
+    // A debugging event.
+    Debug,
+    // An informational event. Indicates that an event happenedate.
+    Info,
+    // A warning event. Not an error but is likely more important than an informational event.
+    Warn,
+    // An error event. Something went wrong.
+    Error,
+    // A fatal error such as application or system crash.
+    Fatal,
+}
+
+/**
+ * A logger for Playground.
+ */
+export class Logger {
+    private readonly LOG_PREFIX = 'Playground';
+
+    /**
+     * The length of the last PHP log.
+     */
+    private lastPHPLogLength = 0;
+
+    constructor() {
+        this.collectPlaygroundLogs();
+    }
+
+    /**
+     * Collect errors from Playground and log them.
+     */
+    private collectPlaygroundLogs() {
+         if (typeof window !== 'undefined') {
+            window.addEventListener('error', (event) => {
+                this.log(
+                    `${event.message} in ${event.filename} on line ${event.lineno}:${event.colno}`,
+                    LogSeverity.Fatal
+                );
+            });
+            window.addEventListener('unhandledrejection', (event) => {
+                this.log(
+                    `${event.reason.stack}`,
+                    LogSeverity.Fatal
+                );
+            });
+            window.addEventListener('rejectionhandled', (event) => {
```

Comment:
Wouldn't this be fired every time a promise is rejected and correctly `caught`? I'm worried about the volume of non meaningful entries it could collect.

Reply:
No, this will be triggered whenever a [rejected Promise is handled late](https://developer.mozilla.org/en-US/docs/Web/API/Window/rejectionhandled_event). They are usually used in combination with `unhandledrejection` for debugging.